Serialiser_Engine/Compute/Serialise/IObject.cs

```diff
@@ -0,0 +1,61 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using MongoDB.Bson.IO;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+
+namespace BH.Engine.Serialiser
+{
+    public static partial class Compute
+    {
+
+        /*******************************************/
+        /**** Public Methods                    ****/
+        /*******************************************/
+        public static void Serialise(this IObject value, BsonDocumentWriter writer)
+        {
+            if (value == null)
+            {
+                writer.WriteNull();
+                return;
+            }
+
+            writer.WriteStartDocument();
+
+            writer.WriteName("_t");
+            writer.WriteString(value.GetType().FullName);
+
+            foreach (PropertyInfo prop in value.GetType().GetProperties())
```

Comment:
Know you know, @adecler and good that it is not done in these first commits, but putting comment for my own memory if nothing else.

Could quite easily optimise this loop by caching `Func<object,object>` for each type being serialised, making use of the `ToFunc()` method in the base engine, and store in a `(Concurrent)Dictionary` with the type or `value.GetType().FullName` as key.

Assume that would speed up serialisation by quite a bit when you have multiple objects of the same type.

Reply:
I know 😄 

In the spirit of keeping things as simple as possible to debug, I have stayed away from that kind of optimisation for now.